zfit_physics/models/pdf_cmsshape.py

```diff
@@ -0,0 +1,68 @@
+from typing import Optional
+
+import numpy as np
+import tensorflow as tf
+import zfit
+from zfit import z
+from zfit.core.space import ANY_LOWER, ANY_UPPER, Space
+from zfit.util import ztyping
+
+
+@z.function(wraps="tensor")
+def cmsshape_pdf_func(x, beta, gamma, m):
+    x = z.unstack_x(x)
+    half = 0.5
+    two = 2.0
+    t1 = tf.math.exp(-gamma * (x - m))
+    t2 = tf.math.erfc(-beta * (x - m))
+    t3 = half * gamma * tf.math.exp(-((half * gamma / beta) ** two))
+    return t1 * t2 * t3
+
+
+@z.function(wraps="tensor")
+def cmsshape_cdf_func(x, beta, gamma, m):
+    half = 0.5
+    two = 2.0
+    y = x - m
+    t1 = tf.math.erf(gamma / (two * beta) + beta * y)
+    t2 = tf.math.exp(-((gamma / (two * beta)) ** two) - gamma * y)
+    t3 = tf.math.erfc(-beta * y)
+    return half * (t1 - t2 * t3) + half
+
+
+def cmsshape_integral(limits: ztyping.SpaceType, params: dict, model) -> tf.Tensor:
+    lower, upper = limits.rect_limits
+    beta = params["beta"]
+    gamma = params["gamma"]
+    m = params["m"]
+    lower_cdf = cmsshape_cdf_func(x=lower, beta=beta, gamma=gamma, m=m)
+    upper_cdf = cmsshape_cdf_func(x=upper, beta=beta, gamma=gamma, m=m)
+    return upper_cdf - lower_cdf
+
+
+class CMSShape(zfit.pdf.BasePDF):
+    _N_OBS = 1
+
+    def __init__(
+        self,
+        beta: ztyping.ParamTypeInput,
+        gamma: ztyping.ParamTypeInput,
+        m: ztyping.ParamTypeInput,
+        obs: ztyping.ObsTypeInput,
+        *,
+        extended: Optional[ztyping.ExtendedInputType] = None,
+        norm: Optional[ztyping.NormInputType] = None,
+        name: str = "CMSShape",
+    ):
+        params = {"beta": beta, "gamma": gamma, "m": m}
+        super().__init__(obs=obs, params=params, name=name, extended=extended, norm=norm)
```

Comment:
Looks good, what about the order of the parameters, i.e. `m`? We should follow as close as possible others, i.e. the voigt.

Reply:
I thought about it. This the the original implementation: https://gitlab.cern.ch/cms-muonPOG/spark_tnp/-/blob/Spark3/RooCMSShape.cc. Here the `alpha` and `peak` are the redundant parameters thet are replaced by just `m`. `m` is a "loc" paraemter and is the approcimate center of the distribution. `scipy` uses the name `loc` for such parameters, i.e. in Gaussian `pdf(x, loc=0, scale=1)` but in `zfit` I see `m` being used. We can put it first if you want and the naming convention is also up to debate.

Reply:
yes, tricky with the names, but as we have m in cauchy, voigt etc, that makes sense to keep it m here too. So m seems reasonable to me. Let's just put it first, so it's consisten with all the other m/mu

Reply:
Done in latest commit.